numpy/core/tests/test_datetime.py

```diff
@@ -1711,23 +1704,18 @@ def test_datetime_arange(self):
         # Unit should be detected as months here
         a = np.arange('1969-05', '1970-05', 2, dtype='M8')
         assert_equal(a.dtype, np.dtype('M8[M]'))
-        assert_equal(a,
-            np.datetime64('1969-05') + np.arange(12, step=2))
```

Comment:
It would be great if we could keep this test somehow... Trying to work out how.

Reply:
If gh-14714 goes in, adding `dtype='m8'` to the arange call would do the trick.

Without it, you need to change `12` to `0, 12` as well.